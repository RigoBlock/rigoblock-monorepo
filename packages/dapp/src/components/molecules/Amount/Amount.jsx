import './Amount.scss'
import ItemValue from '../../atoms/ItemValue'
import PropTypes from 'prop-types'
import React from 'react'
import classNames from 'classnames'

export const AMOUNT_SIZES = {
  SMALL: 'small',
  MEDIUM: 'medium',
  LARGE: 'large'
}

const Amount = ({ value, size, symbol }) => {
  const classProps = classNames('amount', size)
  return (
    <div className={classProps}>
      <div className="amount-symbol">{symbol}</div>
      <ItemValue itemValue={value} valueSize={size} />
    </div>
  )
}

Amount.propTypes = {
  value: PropTypes.object.isRequired,
  symbol: PropTypes.string.isRequired,
  size: PropTypes.oneOf([
    AMOUNT_SIZES.SMALL,
    AMOUNT_SIZES.MEDIUM,
    AMOUNT_SIZES.LARGE
  ])
}

Amount.defaultProps = {
  size: AMOUNT_SIZES.MEDIUM
}

export default Amount
