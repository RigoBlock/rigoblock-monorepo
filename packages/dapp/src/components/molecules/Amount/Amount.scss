@import "../../_settings/variables.scss";

.amount {
  display: flex;
  justify-content: space-between;

  &.small {
    font-size: $medium-font-size;
  }

  &.medium {
    font-size: $large-font-size;
  }

  &.large {
    font-size: $xxl-font-size;
  }
}
