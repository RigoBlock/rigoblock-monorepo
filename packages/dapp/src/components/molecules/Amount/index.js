export { default, AMOUNT_SIZES } from './Amount'
