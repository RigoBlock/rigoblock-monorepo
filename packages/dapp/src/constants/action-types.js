import keyMirror from 'keymirror'

export const actionTypes = keyMirror({
  // Global actions
  GLOBAL_INIT: null,
  // Blockchain actions
  BLOCKCHAIN_INIT: null,
  BLOCKCHAIN_ERROR: null,
  LOGGED_IN: null,
  LOGGED_OUT: null
})
