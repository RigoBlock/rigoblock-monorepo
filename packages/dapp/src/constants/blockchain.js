export const VAULT = 'vault'
export const DRAGO = 'drago'
