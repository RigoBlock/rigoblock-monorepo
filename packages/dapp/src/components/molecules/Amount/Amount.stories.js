import '../../_settings/_base.scss'
import { selectV2, text, withKnobs } from '@storybook/addon-knobs/react'
import { storiesOf } from '@storybook/react'
import Amount, { AMOUNT_SIZES } from './Amount'
import React from 'react'

storiesOf('Molecules/Amount', module)
  .addDecorator(withKnobs)
  .addDecorator(story => (
    <div
      style={{
        width: '250px',
        background: 'white',
        padding: '16px'
      }}
    >
      {story()}
    </div>
  ))
  .add('default', () => (
    <Amount
      size={selectV2('Amount size', AMOUNT_SIZES, AMOUNT_SIZES.SMALL)}
      symbol={text('Amount symbol', 'ETH')}
      value={8000.23}
    />
  ))
