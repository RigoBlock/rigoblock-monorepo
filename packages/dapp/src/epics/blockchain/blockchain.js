import 'rxjs/add/operator/do'
import 'rxjs/add/operator/merge'
import 'rxjs/add/operator/mergeMap'
import { Scheduler } from 'rxjs/Scheduler'
import { Subject } from 'rxjs/Subject'
import { actionTypes } from '../../constants/action-types'
import { of } from 'rxjs/observable/of'
import BlockChainService from './BlockChainService'
import api from '../../api'
import blockChainActions from '../../actions/blockchain-actions'

const blockchainSubject = new Subject()
let blockchainService

export const blockchainEpic = (action$, store, ts = Scheduler.async) => {
  blockchainService = new BlockChainService(api, action$, blockchainSubject, ts)
  return action$
    .filter(action => action.type === actionTypes.GLOBAL_INIT)
    .mergeMap(() => {
      return window.web3
        ? blockchainService.init().merge(blockchainSubject)
        : of(blockChainActions.blockChainLogout())
    })
}

export const blockchainFetchEventsEpic = action$ => {
  return action$
    .filter(action => action.type === actionTypes.LOGGED_IN)
    .mergeMap(() => blockchainService.fetchVaultEvents())
}

export default [blockchainEpic, blockchainFetchEventsEpic]
