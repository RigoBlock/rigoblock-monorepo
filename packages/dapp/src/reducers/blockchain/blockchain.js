import { actionTypes } from '../../constants/action-types'
import updeep from 'updeep'
import vaultsReducer from './vaults'

const initialState = {
  accounts: {}
}

function blockChainReducer(state = initialState, action) {
  switch (action.type) {
    case actionTypes.LOGGED_IN:
      return state.accounts[action.payload.account]
        ? state
        : updeep(state, {
            accounts: { [action.payload.account]: {} }
          })
    case actionTypes.ADD_VAULT:
      return updeep(state, {
        accounts: {
          [action.payload.account]: { vaults: vaultsReducer(undefined, action) }
        }
      })
    default:
      return state
  }
}

export default blockChainReducer
