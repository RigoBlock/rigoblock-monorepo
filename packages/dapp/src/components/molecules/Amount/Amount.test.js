import { BigNumber } from 'bignumber.js'
import { ETH } from '../../../constants/blockchain'
import { shallow } from 'enzyme'
import Amount from './Amount.jsx'
import toJson from 'enzyme-to-json'

const props = { value: new BigNumber('8000.23'), symbol: ETH }

describe('Amount component', () => {
  it('renders correctly', () => {
    expect(
      toJson(shallow(createComponentWithProps(Amount, props)))
    ).toMatchSnapshot()
  })
})
