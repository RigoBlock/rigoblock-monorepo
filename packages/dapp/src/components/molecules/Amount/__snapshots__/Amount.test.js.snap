// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Amount component renders correctly 1`] = `
<div
  className="amount"
>
  <div
    className="amount-symbol medium"
  >
    ETH
  </div>
  <ItemValue
    currency="ETH"
    currencyGrowth={null}
    growth={null}
    itemValue={"8000.23"}
    precision={2}
    valueSize="medium"
  />
</div>
`;
